### PR TITLE
BOM-2928: Replace Travis with GitHub CI

### PR DESCRIPTION
**Pull request by UsamaSadiq**
**TL;DR -** [ Replacing Travis with GitHub CI ]

JIRA: [JIRA-2928](https://openedx.atlassian.net/browse/JIRA-2928)

**What changed?**

- [ Removed Travis.yml ]
- [ Added GitHub CI for tests and deployment ]
- [ Removed greenkeeper lock since it has been [deprecated](https://github.com/greenkeeperio/greenkeeper-lockfile#readme) and doesn't have support for GitHub CI]

**Developer Checklist**
- [x] Test suites passing
- [x] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [x] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ CI tests should pass. ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Review by mraarif — APPROVED** [review]

File: documentation/CI.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+1. Definitely make it really verbose so people can't remember what it's called
+2. Definitely don't not use a double-negative
+
+What `make validate-no-uncommitted-package-lock-changes` does is `git diff`s for any `package-lock.json` file changes in your project. It's important to remember that all build `script`s are executed in CI *after* the `install` step (aka post-`npm install`).
+
+This is important because `npm` uses the pinned dependencies in your `package-lock.json` file to build the `node_modules` directory. However, the dependencies defined within the `package.json` file can be modified manually, for example, to become misaligned with the dependencies defined within the `package-lock.json`. So when `npm install` executes, the `package-lock.json` file will be updated to mirror the modified `package.json` changes.
+
+However, when these changes surface within a CI build, this indicates differing dependency expectations between the committed `package.json` file and the `package-lock.json` file, which is a good reason to fail a build.
+
+### What is this `npm run is-es5` check?
```
Comment (leangseu-edx) [inline documentation/CI.md:28]: 👍

**Review by leangseu-edx — APPROVED** [review]
LGTM

**Review by mattcarter — APPROVED** [review]
Looks good to me 👍
